### PR TITLE
UI search bar - search conditions ignored #LMR-611

### PR DESCRIPTION
**Pull request by kubedo8**

### CODE REVIEW

File: src/app/core/store/navigation/query.model.ts [hunk 1]
```diff
@@ -32,3 +32,44 @@ export interface QueryModel {
...
+}
+
+const EqVariants = ['=', '==', 'eq', 'equals'];
+const NeqVariants = ['!=', '!==', '<>', 'ne', 'neq', 'nequals'];
+const LtVariants = ['<', 'lt'];
+const LteVariants = ['<=', 'lte'];
+const GtVariants = ['>', 'gt'];
+const GteVariants = ['>=', 'gte'];
+
+export function conditionFromString(condition: String): ConditionType {
```
Comment (livthomas) [inline src/app/core/store/navigation/query.model.ts]: I can smell some Java here again: `string` :)

File: src/app/core/store/documents/documents.filters.ts [hunk 2]
```diff
@@ -18,50 +18,89 @@
...
+    .filter(filter => !isNullOrUndefined(filter));
 
-    return documents.filter(document => Object.values(document.data).some(value => value.includes(query.fulltext)));
+  return documents.filter(document => documentMeetsFilters(document, filters));
+}
+
+function documentMeetsFilters(document: DocumentModel, filters: AttributeFilter[]): boolean {
+  let meets = true;
+  filters.forEach(filter => meets = meets && documentMeetFilter(document, filter));
+  return meets;
```
Comment (livthomas) [inline src/app/core/store/documents/documents.filters.ts]:
```typescript
return filters.every(filter => documentMeetFilter(document, filter));
```

File: src/app/core/store/navigation/query.model.ts [hunk 3]
```diff
@@ -32,3 +32,44 @@ export interface QueryModel {
   not?: QueryModel;
 
 }
+
+export interface AttributeFilter {
+  collectionId: string,
+  conditionType: ConditionType;
+  attributeName: string;
```
Comment (livthomas) [inline src/app/core/store/navigation/query.model.ts]: I assume you mean `attributeId` here.

File: src/app/core/store/documents/documents.filters.ts (hunk 2 above)
Comment (kubedo8) [reply src/app/core/store/documents/documents.filters.ts]: I was looking for something like that, thanks!

**Comment by kubedo8** [comment]
@livthomas fixed.

**Pull request merged**